### DISCUSSION TITLE
Add option to rename Workflow runs

### DISCUSSION THREAD

**Discussion by jiribrejcha** [community/community#64836] [Actions]
### Select Topic Area

Product Feedback

### Body

Currently, all Workflow runs are named after the Workflow. I am suggesting we add a "Rename" option to GitHub Workflows, which will allow us to add a note to the Workflow run name. For example "Do not use", or "Built from dev branch", "Kernel 6.1" or similar comments.

Here is an example of our Workflow. All runs are called "Build kernel", but they are different flavours and Kernel versions. Ability to rename them would save us a ton of time.

![image](https://github.com/community/community/assets/18701525/35c8452a-981d-4c6a-8625-970ce311703d)

**Comment by deadlydog** [comment]
You can use the [run-name](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#run-name) element to dynamically set the workflow run name. It is very limited though as it is only able to access the `github` and `inputs` contexts.

I came here to make a similar request as this, to be able to rename the workflow run from a job step. My use case is we want to include the version number (which is determined dynamically during the build workflow run) in our build and deployment workflow run names. Having to click into every workflow run to find out which version it is for (assuming you output it in one of your workflow steps) is a real pain.

+1 to be able to include arbitrary information in the workflow run name from a job step.

**Reply by zeckem19** [reply]
adding a version number into workflow run would be a great feature

**Reply by utsabshresthajw** [reply]
Hi @deadlydog have you got any workaround solution for getting the build version info without having to look into workflow steps? Please let me know. Thanks!

**Reply by deadlydog** [reply]
> Hi @deadlydog have you got any workaround solution for getting the build version info without having to look into workflow steps? Please let me know. Thanks!

Nope, still waiting on the team to implement a solution.

**Reply by utsabshresthajw** [reply]
@deadlydog Thanks for the reply

**Reply by Vaccano** [reply]
Three years later, and still no luck on this?  I truly thought GitHub Actions to be more mature that this! :(

**Comment by deadlydog** [comment]
Reading through the original post again, I think the OP is requesting the ability to rename a workflow run _that has already completed_. That is different from my request. I want the ability to specify the run name during one of the job steps, which is #69476.

@jiribrejcha you may want to update the title of this request to make that distinction clear. e.g.  Change the title to `Add option to rename completed Workflow runs`.

**Comment by frackham** [comment]
This'd still be great to see. 

Really useful for multiple related composite runs, where you may want the run to show as success/fail as normal but surface some information from the child composite action

**example scenario**: Workflow called `Orchestrate`, only ever manually triggered so name will always be 'Orchestrate' unless changed via run-name, using a 'determine next step' composite action will always take an input and return an output step. Nice, simple modular workflow as the next step may be used elsewhere.
**as-is**: We can use `run-name` to set the run name at runtime using context/inputs, but only in the composite, so will never be able to surface the output in the parent `Orchestrate` workflow's run-name. _Result is that the list of runs under Orchestrate will all be called `Orchestrate`._
**to-be**: Once the `Orchestrate` workflow receives a response/output from the composite, the run-name is appended with the state changes. _Result is that the list of runs under Orchestrate can be called `Orchestrate (From INITIAL_STATE to NEW_STATE)` with dynamic values of the states, meaning you could see transitions at a glance in the outputs._
